packages/react-vapor/src/components/modalWizard/ModalWizard.tsx

```diff
@@ -0,0 +1,108 @@
+import classNames from 'classnames';
+import * as React from 'react';
+import {connect} from 'react-redux';
+
+import {ConnectedProps, IDispatch, TooltipPlacement} from '../../utils';
+import {Button} from '../button';
+import {ModalActions} from '../modal/ModalActions';
+import {IModalCompositeOwnProps, ModalCompositeConnected} from '../modal/ModalComposite';
+import {UnsavedChangesModalProvider} from '../modal/UnsavedChangesModalProvider';
+import {StepProgressBar} from '../stepProgressBar';
+
+export interface ModalWizardProps
+    extends Omit<IModalCompositeOwnProps, 'modalBodyChildren' | 'validateShouldNavigate'> {
+    id: string;
+    onFinish?: () => unknown;
+    validateStep?: (currentStep: number) => string[];
+    isDirty?: boolean;
+    cancelButtonLabel?: string;
+    nextButtonLabel?: string;
+    previousButtonLabel?: string;
+    finishButtonLabel?: string;
+}
+
+const enhance = connect(null, (dispatch: IDispatch, {id}: ModalWizardProps) => ({
+    close: () => dispatch(ModalActions.closeModal(id)),
+}));
+
+const ModalWizardDisconneted: React.FunctionComponent<ModalWizardProps & ConnectedProps<typeof enhance>> = ({
```

Comment:
I know that you did not want it to be bound to the Validations feature, but could we provide an HOC to enhance `ModalWizard` with the `validationIdsByStep` prop? It would replace us having to define `isDirty` and `validateStep`.

Reply:
Implementing it in a separate PR